util/blobs/blobs.go

```diff
@@ -47,6 +79,20 @@ func DecodeBlobs(blobs []kzg4844.Blob) ([]byte, error) {
 		for fieldIndex := 0; fieldIndex < params.BlobTxFieldElementsPerBlob; fieldIndex++ {
 			rlpData = append(rlpData, blob[fieldIndex*32+1:(fieldIndex+1)*32]...)
 		}
+		var acc uint16
+		accBits := 0
+		for fieldIndex := 0; fieldIndex < params.BlobTxFieldElementsPerBlob; fieldIndex++ {
+			acc |= uint16(blob[fieldIndex*32]) << accBits
+			accBits += spareBlobBits
+			if accBits >= 8 {
+				rlpData = append(rlpData, uint8(acc))
+				acc >>= 8
+				accBits -= 8
+			}
+		}
+		if accBits != 0 {
```

Comment:
nice, this should never happen because 4096 is divisible by 4 and the pattern for accBits is 2, 4, 6, 0, repeat on encoding and 6, 4, 2, 0, repeat on decoding.

Reply:
Exactly, though this condition (well the one on encoding) helped track down a bug before where I exited the loop early and left some bits in the accumulator